components/hab/src/error.rs

```diff
@@ -127,6 +128,13 @@ impl fmt::Display for Error {
             Error::HabitatCore(ref e) => format!("{}", e),
             Error::HandlebarsRenderError(ref e) => format!("{}", e),
             Error::IO(ref err) => format!("{}", err),
+            Error::JobGroupPromote(ref id, ref c) => {
+                format!(
+                    "Failed to promote job group ID {} to channel {} because the job group is not complete.",
```

Comment:
There is an enum in the `scheduler.proto` file that lists out all the states a project could be in.  None of them are exposed to users.  I'll update this text.